bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/extension/Extension.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.extension;
+
+/**
+ * This class defines an extension.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
+ */
+public class Extension {
```

Comment:
I changed it to an immutable object with getter methods and a constructor.

> Should we rename type in typeId?

I thought about this, but as the class is used as a DTO for JSON on REST, I named it type, since this is what we do in JSON structures usually.